### ISSUE TITLE
Timezone issue and new filters for API /wordpress/wp-json/wc/v1/orders

### ISSUE THREAD

**Issue by gkwang0** [woocommerce/woocommerce#12594]
There is one issue about timezone when I testing /wordpress/wp-json/wc/v1/orders. The plugin version is 2.6.9. The issue happened after I set the timezone of Wordpress to UTC-1 at Wordpress Settings page. I save the date_created of previous order in my local and use it as parameter "after"to search the new orders in new request, therefore no need always get all orders. But the parameter "after" is not used as UTC time make me confused. The steps to reproduce this issue are as below:
1. Set the timezone of Wordpress to UTC-1 at Wordpress Settings page
2. Create a new order on WooCommerce system at the time 2016-12-13T04:33:33(UTC-1)
3. Get the orders without parameter "after", and all order are returned, date_created of the order in response is 2016-12-13T05:33:33(UTC), I think this is correct.
4. Create a new order on WooCommerce system again, for example the time 2016-12-13T04:35:00(UTC-1), date_created of the order should be 2016-12-13T05:35:00(UTC)
5. Get the orders with using "2016-12-13T05:33:33" for parameter "after", but the second order is not returned. I need using the UTC-1 time "2016-12-13T04:33:33" then I can get the second order. 

**I think it should using parameter "after" as UTC time to do searching, not using it as the time zone of Wordpress.** Will this issue be fixed?

The other issues about the API /wordpress/wp-json/wc/v1/orders
1. The "after" parameter is used to search the orders created after the time, Is it possible to add additional parameters for searching the orders modified after the time?
2. I can using "date" as the value of "orderby" parameter to search the orders sorting by the created time, Is it possible to add additional enumeration for "orderby" for searching the orders sorting by the modified time?

**Comment by mikejolley** [comment]
After talking with Claudio, date_created follows the way WP api does things (timezone relative). Converting to UTC times would be incompatible with that.

If the WP Rest API used UTC, we would too, but thats not the case.

Looks like you'll soon be able to at least get the offset via API in future versions though which could be useful https://core.trac.wordpress.org/ticket/39072

Note: "after" is provided by the WP Rest API, not WC directly.

**Comment by claudiosanches** [comment]
> The "after" parameter is used to search the orders created after the time, Is it possible to add additional parameters for searching the orders modified after the time?

Not until WordPress core don't have something like it.

> I can using "date" as the value of "orderby" parameter to search the orders sorting by the created time, Is it possible to add additional enumeration for "orderby" for searching the orders sorting by the modified time?

Another thing that we are doing like WordPress core.

**Comment by gkwang0** [comment]
@mikejolley @claudiosanches 
I investigated the Woocommerce REST API query parameters and the WP query parameters, and I use the "filter" parameter of Woocommerce REST API to do searching for now and it can resolve my issues. 

About the "after" parameter is regard as the time related to setting timezone not UTC time to do search, I think this is caused by there is not query parameters for which date time column is searched for, and it default to search the order whose "post_date" is after parameter and this column is related to the timezone. I used "filter[date_query][0][after]=yyyy-MM-dd HH:mm:ss" and "filter[date_query][0][column]=post_date_gmt" to invoke API and it can search the orders as expected. 

About searching the order whose modified time is after the parameter, I used "filter[date_query[0][after]=yyyy-MM-dd HH:mm:ss" and "filter[date_query][0][column]=post_modified_gmt" to do searching.

Searching the order sorting by the modified time, I used "filter[order]=modified" can get the orders sorting by modified time correctly.

Hope my temporary solution is helpful.
